### DISCUSSION TITLE
Issue with 11.1 Web Adaptor dependancies

### DISCUSSION THREAD

**Discussion by geoprofessor** [Esri/arcgis-powershell-dsc#493] [General]
Hello,

While performing a 10.9.1->11.1 upgrade from stand alone ArcGIS Server w/ web adaptor. The install/upgrade fails at the Web Adaptor without any good error messages. I ran the IIS web adaptor installer manually and it appears it requires Web Deploy 3.6 and .NET hosting support. After installing these manually the rest of the upgrade process went thru smoothly.

During the powershell upgrade of the web adaptor I see it checking other IIS dependencies, is there a reason these new ones are not included? 

Thanks

**Comment by cameronkroeker** [comment]
Hi @geoprofessor,

The following attributes can be defined in your json configuration file to install the Web Deploy and Dot Net Hosting Bundle during fresh deployments and upgrades:

<img width="884" alt="Screenshot 2023-09-05 at 2 05 15 PM" src="https://github.com/Esri/arcgis-powershell-dsc/assets/48299621/49513ecb-9b0c-481a-b022-64ebba9b6fe3">

Sample Json:
https://github.com/Esri/arcgis-powershell-dsc/blob/8621f7a7256d6f8b9431caf4674e01c8d3bce249/SampleConfigs/v4/v4.1.0/Base%20Deployment/BaseDeployment-SingleMachine.json#L171-L174

If these attributes are not defined in your json configuration file then it is assumed that these were installed prior to running the ArcGIS Module. I do agree, however, that it would be a better experience to have some sort of validation check in place for it check to see if Web Deploy/Dot Net Hosting Bundle exists and if doesn't throw a useful error message. Maybe this is something we can take a look at in a future release.

Thanks,
Cameron K.

**Reply by geoprofessor** [reply]
Thanks @cameronkroeker  this is very helpful!
